File: chazm-model/src/main/java/runtimemodels/chazm/model/event/PossessesEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.relation.Possesses;

/**
 * The {@linkplain PossessesEventFactory} interface defines the API for constructing {@linkplain PossessesEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PossessesEventFactory {

    /**
     * Constructs an {@linkplain PossessesEvent}.
     *
     * @param category  the {@linkplain EventType}.
     * @param possesses the {@linkplain Possesses}.
     * @return a {@linkplain PossessesEvent}.
     */
    PossessesEvent build(EventType category, Possesses possesses);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/PossessesEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.relation.Possesses

/**
 * The [PossessesEventFactory] interface defines the API for constructing [PossessesEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PossessesEventFactory {

    /**
     * Constructs an [PossessesEvent].
     *
     * @param category  the [EventType].
     * @param possesses the [Possesses].
     * @return a [PossessesEvent].
     */
    fun build(category: EventType, possesses: Possesses): PossessesEvent

}
